Tools/Helpers/carbon.sh

```diff
@@ -5,6 +5,6 @@
 ### All rights reserved
 ###
 
-SCRIPT=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-source "${SCRIPT}/carbon/tools/environment.sh"
-"${SCRIPT}/RustDedicated" "$@"
\ No newline at end of file
+CARBON_INIT_SCRIPT=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source "${CARBON_INIT_SCRIPT}/carbon/tools/environment.sh"
+"${CARBON_INIT_SCRIPT}/RustDedicated" "$@"
```

Comment:
I would agree with this if the variable weren't named so abstractly.
A variable with this name can be defined in the global environment, for example, as an initialization script for a docker container, and will be overwritten with a new value inside the current script, what could break next sequences.

Reply:
The script `carbon.sh` is supposed to be an entry-point, so it's fine for the var to be "generic" as we have control over the context; on the other hand `environment.sh` is supposed to be invoked from other scripts, so we can't guarantee the starting context.

Reply:
Yes, `carbon.sh` is the entry-point to the server and the mod framework itself, but there could exist some additional user scripts, that provide `steamcmd`'s server update automatization, mod updates, failover restarts in case of the server fails and so on. Those scripts will run the server entry point, watch run and retries on fails, and it will suddenly fail during the first restart because of unexpected data change of the variable.
It will work fine if such scripts will run bare server, but when they switch to use Carbon, and "something" rewrites variables inside user scripts, it causes a reaction "Whoat?" and user have to dig deeper to understand what is actually going on.

I understand, that `environment.sh` setup their own environment variables, that required for the framework work, and that is expected behavior. But that is unexpected, if the entry-point script will behave as the environment setup also, otherwise this behavior should be documented, so users won't use the same var names as in your scripts in their utility scripts. Or better way, is your scripts should contain unique var names that will have the least impact on the user environment.

Reply:
Whatever happens upstream we don't care, only downstream. You can run steam cmd, scripts, go to the moon.. the condition triggering the error can only happen downstream of `carbon.sh`. Try it, create a bash script, set the var and then invoke `carbon.sh`.

Reply:
Amazing responsibility.
You can close the pull request without merging.

Reply:
This has nothing to do with responsibility. The reported error is a valid error, but your proposed fix is going further than required for no reason.